crates/metal-ast-ng/metal.ungram

```diff
@@ -0,0 +1,153 @@
+Name =
+  '@ident'
+
+Visibility =
+  'pub'?
+
+Mutability =
+  'mut'?
+
+ExprSpecifier =
+  '=' Expr
+
+TypeQualifier =
+  ':' Type
+
+Block =
+  '{' Item '}'
+
+Item =
+  ConstItem | EnumItem | Expr | FnItem | ImportItem | StructItem
+
+ConstItem =
+  vis:Visibility 'const' Name ty:TypeQualifier? value:ExprSpecifier
+
+EnumItem =
+  vis:Visibility 'enum' Name '{' body:EnumBody? '}'
+
+EnumBody =
+  items:(EnumBodyItem (',' EnumBodyItem)* ','?)
+
+EnumBodyItem =
+  EnumVariant | FnItem
+
+EnumVariant =
+  Name data_ty:EnumVariantType?
+
+EnumVariantType =
+  ('(' Type ')')
+
+FnItem =
+  vis:Visibility 'def' Name sig:FnSignature body:Block
+
+FnSignature =
+  '(' inputs:FnInputs? ')' return_ty:TypeQualifier?
+
+FnInputs =
+  FnInput (',' FnInput)* ','?
+
+FnInput =
+  mutness:Mutability Name ty:TypeQualifier? default:ExprSpecifier?
+
+ImportItem =
+  vis:Visibility 'import' tree:ImportTree ';'
+
+ImportTree =
+  ImportLeaf | ImportSegment | ImportBranch
+
+ImportLeaf =
+  Name
+
+ImportSegment =
+  segment:Name '.' rest:ImportTree
+
+ImportBranch =
+  '{' subtrees:ImportBranchSubtrees? '}'
+
+ImportBranchSubtrees =
+  ImportTree (',' ImportTree)* ','?
+
+StructItem =
+  vis:Visibility 'struct' Name '{' body:StructBody '}'
+
+StructBody =
+  items:(StructBodyItem (',' StructBodyItem)* ','?)
+
+StructBodyItem =
+  StructField | FnItem
+
+StructField =
+  vis:Visibility Name ty:TypeQualifier
+
+Type =
```

Comment:
It seems this does no AST interpretation of the inner types? (Vec[T])

Reply:
Generics are indeed not in the grammar

Reply:
Done!